File: Anagrams/answer.js
var anagrams = function(strs) {
  var ret = [];
  var obj = {};

  strs.forEach(function(str, i) {
    str = str.split('').sort().join('');
    if (!obj[str]) {
      obj[str] = [strs[i]];
    } else {
      obj[str].push(strs[i]);
    }
  });

  Object.keys(obj).forEach(function(str) {

    if (obj[str].length > 1) {
      ret = ret.concat(obj[str]);
    }
  });

  return ret;
};

File: Anagrams/answer.py
class Solution:
    # @param strs, a list of strings
    # @return a list of strings
    def anagrams(self, strs):
        
        dict = {}
        ret = []
        for str in strs:
            newstr = ''.join(sorted(str))

            if newstr not in dict:
                dict[newstr] = [str]
            else:
                dict[newstr].append(str)

        for key in dict:
            if len(dict[key]) >= 2:
                ret += dict[key]
        return ret


